main.js

```diff
@@ -1,163 +1,277 @@
+            hdr: true,
+            bluetooth: false,
+            ambiLight: true,
+        },
+        originalStock: 6,
+        sold: 3,
     },
-    originalStock: 6,
-    sold: 3,
-  },
-  {
-    type: 'B2450HD',
-    name: 'LED TV',
-    brand: 'Brandt',
-    price: 109,
-    availableSizes: [24],
-    refreshRate: 60,
-    screenType: 'LED',
-    screenQuality: 'Full HD',
-    smartTv: false,
-    options: {
-      wifi: false,
-      speech: false,
-      hdr: false,
-      bluetooth: false,
-      ambiLight: false,
+    {
+        type: 'B2450HD',
+        name: 'LED TV',
+        brand: 'Brandt',
+        price: 109,
+        availableSizes: [24],
+        refreshRate: 60,
+        screenType: 'LED',
+        screenQuality: 'Full HD',
+        smartTv: false,
+        options: {
+            wifi: false,
+            speech: false,
+            hdr: false,
+            bluetooth: false,
+            ambiLight: false,
+        },
+        originalStock: 10,
+        sold: 8,
     },
-    originalStock: 10,
-    sold: 8,
-  },
-  {
-    type: '32WL1A63DG',
-    name: 'HD TV',
-    brand: 'Toshiba',
-    price: 161,
-    availableSizes: [32],
-    refreshRate: 50,
-    screenType: 'LED',
-    screenQuality: 'Full HD',
-    smartTv: false,
-    options: {
-      wifi: false,
-      speech: false,
-      hdr: true,
-      bluetooth: false,
-      ambiLight: false,
+    {
+        type: '32WL1A63DG',
+        name: 'HD TV',
+        brand: 'Toshiba',
+        price: 161,
+        availableSizes: [32],
+        refreshRate: 50,
+        screenType: 'LED',
+        screenQuality: 'Full HD',
+        smartTv: false,
+        options: {
+            wifi: false,
+            speech: false,
+            hdr: true,
+            bluetooth: false,
+            ambiLight: false,
+        },
+        originalStock: 10,
+        sold: 8,
     },
-    originalStock: 10,
-    sold: 8,
-  },
-];
\ No newline at end of file
+];
+//1a
+const tvsStockMapper = inventory.map((stock) => {
+    return stock.originalStock - stock.sold
+});
+const totalTvsInStock = () => {
+    let allStock = 0
+    for (let i = 0; i < inventory.length; i++) {
+        allStock = allStock + tvsStockMapper[i];
+    }
+    return allStock
+}
+// 1b
+const stockField = document.getElementById('current-stock');
+stockField.innerText = `Totale huidige voorraad: \n${totalTvsInStock().toString()} stuks.`
+//  2a
+const tvTypeMapper = inventory.map((type) => {
+    return type.type;
+});
+//  2b
+const soldOut = inventory.filter((sold) => {
+    return sold.originalStock - sold.sold === 0;
+});
+//  2c
+const ambiLight = inventory.filter((ambi) => {
+    return ambi.options.ambiLight;
+});
+//  2d (aka 2e)
+const priceSort = inventory.sort((a, b) => a.price - b.price);
+// 3a
+const price = inventory.map((prijs) => {
+    return prijs.price;
+});
+let totalRevenue = 0;
+for (let i = 0; i < inventory.length; i++) {
+    totalRevenue += tvsStockMapper[i] * price[i];
+}
```

Comment:
Ik zie liever dat je toch dezelfde array gebruikt i.p.v. twee verschillende. In dit geval gaat het goed, maar als je één van de twee anders zou sorteren of filteren dan werkt het meteen niet meer. Wat je zou kunnen doen:
```
let totalRevenue = 0;
for (let i = 0; i < inventory.length; i++) {
    totalRevenue += inventory[i].originalStock * inventory[i].price;
}
```
Of nog korter:
```
const totalRevenue = inventory.reduce((total, tv) => (total += tv.originalStock * tv.price), 0);
```